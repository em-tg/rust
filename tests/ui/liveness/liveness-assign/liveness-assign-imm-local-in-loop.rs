//@ check-pass
fn test() {
    let v: isize;
    //~^ HELP consider making this binding mutable
    //~| SUGGESTION mut
    loop {
        v = 1; //~ WARNING cannot assign twice to immutable variable `v`
               //~| NOTE cannot assign twice to immutable variable
               //~| NOTE on by default
    }
}

fn main() {
}
