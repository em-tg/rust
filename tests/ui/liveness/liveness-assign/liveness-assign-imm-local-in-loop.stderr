warning[E0384]: cannot assign twice to immutable variable `v`
  --> $DIR/liveness-assign-imm-local-in-loop.rs:7:9
   |
LL |         v = 1;
   |         ^^^^^ cannot assign twice to immutable variable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider making this binding mutable
   |
LL |     let mut v: isize;
   |         +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0384`.
